### PR TITLE
Improve pidfile management

### PR DESCRIPTION
**Pull request by nirs**
We used to create an empty pidfile, and write the pid only after starting the interface. Having pidfile without a pid does not make sense, as other processes cannot use the pid to terminate the process. For example if starting the interface is blocking for long time due to a bug, we are left with unhelpful pid file.

To keep the main flow more clear, add a helper to create and remove the pid file. The helpers check all errors and log better log messages using ERRORF instead of ERRORN, including the pidfile path.

Previously if we got a short write to the pidfile, we log bogus error since errno is set only when write return negative return value.

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -364,6 +364,40 @@ static int socket_bindlisten(const char *socket_path,
...
+{
+  if (unlink(pidfile) != 0) {
+    ERRORF("Failed to remove pidfile: \"%s\": %s", pidfile, strerror(errno));
+  }
+}
+
+static int create_pidfile(const char *pidfile)
+{
+  int flags = O_WRONLY | O_CREAT | O_EXLOCK | O_TRUNC | O_NONBLOCK;
+  int pid_fd = open(pidfile, flags, 0644);
```
Comment (AkihiroSuda) [inline main.c:377]: nit:
```suggestion
  int fd = open(pidfile, flags, 0644);
```

To me, `pid_fd` sounds like Linux's pidfd https://man7.org/linux/man-pages/man2/pidfd_open.2.html

Comment (nirs) [reply main.c:377]: Yes, I also thought about pidfd. We also rename it to pidfile_fd in main to avoid the this confusion.

Comment (nirs) [reply main.c]: Renamed

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**